src/HandHistory.js

```diff
@@ -0,0 +1,148 @@
+class HandHistory {
+
+    constructor(player, cards) {
+        this.actions = [];
+        this.hero = player;
+        this.heroCards = cards;
+    }
+
+    // After you build one then for fuck's sake don't modify it. I'm trusting you.
+
+    // Actions
+
+    addStack(player, amount) {
+        this.actions.push({action: 'stack', player: player, amount: amount});
+    }
+
+    addBet(player, amount) {
+        this.actions.push({action: 'bet', player: player, amount: amount});
+    }
+
+    fold(player) {
+        this.actions.push({action: 'fold', player: player});
+    }
+
+    check(player) {
+        this.actions.push({action: 'check', player: player});
+    }
+
+    dealFlop(cards) {
+        this.actions.push({action: 'flop', cards: cards});
+    }
+
+    dealTurn(card) {
+        this.actions.push({action: 'turn', card: card});
+    }
+
+    dealRiver(card) {
+        this.actions.push({action: 'river', card: card});
+    }
+
+    actionOn(player) {
+        this.actionOn = player;
+    }
+
+    // Queries
+
+    stack(player) {
+        var result = 0;
+        this.actionsFor(player).forEach(each => {
+            switch(each.action) {
+                case 'stack':
+                    result += each.amount;
+                    break;
+                case 'bet':
+                    result -= each.amount;
+                    break;
+            }
+        });
+        return result;
+    }
+
+    pot() {
```

Comment:
One bit of javascript class syntax: if you wanted to have these look like properties rather than methods, you can define them with the `get` keyword and then call them without the parentheses. Totally unnecessary, and here it might be more relevant to keep the symmetry with the functions that ask a question about a specific player...